assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,300 @@
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+    outline: none;
+  }
+`;
+
+function DropDown({ options, value, onChange, disabled }) {
+  DropDown.wrapperRef = useRef(null);
+  DropDown.selectRef = useRef();
+  DropDown.arrayOfOptionsRefs = [];
+
+  const [openOptions, setOpenOptions] = useState(false);
+  const [focusedOption, setFocusedOption] = useState(undefined);
+  const [searchValue, setSearchValue] = useState('');
+  const activeItem = useMemo(
+    () => options.find((item) => item.value === value),
+    [value, options]
+  );
+  const toggleOptions = useCallback(() => setOpenOptions(false), []);
+
+  useFocusOut(DropDown.wrapperRef, toggleOptions);
+  useKeyDownEffect(DropDown.wrapperRef, { key: 'esc' }, toggleOptions);
+
+  const clearOptionsRefs = () => {
+    DropDown.arrayOfOptionsRefs = [];
+  };
+
+  const handleOpenOptions = (event) => {
+    switch (event.type) {
+      case 'click':
+        setOpenOptions(!openOptions);
+        setFocusedOption(document.activeElement.id);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          _handleOpenOptions(event);
+        }
+        break;
+      default:
+    }
+  };
+
+  const _handleOpenOptions = useCallback(() => {
+    setOpenOptions(!openOptions);
+  }, [openOptions]);
+
+  const handleCurrentValue = useCallback(
+    (option) => {
+      if (onChange) {
+        onChange(option);
+      }
+    },
+    [onChange]
+  );
+
+  const handleSelectElement = (event, index) => {
+    event.preventDefault();
+    if (index >= 0 && index < DropDown.arrayOfOptionsRefs.length) {
+      DropDown.arrayOfOptionsRefs[index].focus();
+      setFocusedOption(document.activeElement.id);
+    }
+  };
+
+  const handleOptionsEvents = (option, index, event) => {
+    switch (event.type) {
+      case 'click':
+        handleCurrentValue(option);
+        setOpenOptions(!openOptions);
+        break;
+      case 'keydown':
+        if (event.key === 'Enter' || event.key === ' ') {
+          handleCurrentValue(option);
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else if (event.key === 'ArrowUp') {
+          handleSelectElement(event, index - 1);
+        } else if (event.key === 'ArrowDown') {
+          handleSelectElement(event, index + 1);
+        } else if (event.key === 'Escape') {
+          event.preventDefault();
+          setOpenOptions(!openOptions);
+          DropDown.selectRef.current.focus();
+        } else {
+          const searchTerm = searchValue + String.fromCharCode(event.keyCode);
+          setSearchValue(searchTerm);
+          clearSearchValue();
```

Comment:
I know that this doesn't easily translate to `useKeyDownEffect` (using Mousetrap behind the scenes, that [doesn't support character classes](https://github.com/ccampbell/mousetrap/issues/134)), but I still want to use it for this. Let's figure out a better way to do this.

Reply:
Currently only using a to z and 1 to 9 and only plus the string.